### ISSUE TITLE
How does the --warm_start param work?

### ISSUE THREAD

**Issue by EuphoriaCelestial** [NVIDIA/tacotron2#328]
I am training a new Tacotron model from sratch. The first run I use this command: 
`python train.py --output_directory=outdir --log_directory=logdir`
and run to 200th iter then stop. Now I want to resume the training, but I notice that there is a `--warm_start` param for train command, so I tried:
`python train.py --output_directory=outdir --log_directory=logdir -c outdir/checkpoint_200 --warm_start`
but it train from iter 0 again, and new checkpoint files saved from 0 not from 200 as I expected. And if I dont use `--warm_start` it will continue train from last iter, and new checkpoint file number increased (300, 400, ....)

**Comment by ksaidin** [comment]
Hello @EuphoriaCelestial, when checkpoint is loaded without warm_start, it gets iteration from checkpoint_dict and starts incrementing from that number.

The coolest difference is that warm start can ignore embedding weights of pretrained model/checkpoint. So that anyone can warm start from [this tacotron 2 model](https://drive.google.com/file/d/1c5ZTuT7J08wLUoVZ2KkUs_VdZuJ86ZqA/view?usp=sharing) and get alignment image even after 500 iterations. Thank you @rafaelvalle !

Now I am trying to understand how learning rate is updated, when I resume my training without warm start, it seems it gets learning rate from checkpoint_dict. 

Do I change ignore_layers and warm start in order to update my learning rate?

**Comment by EuphoriaCelestial** [comment]
> Hello @EuphoriaCelestial, when checkpoint is loaded without warm_start, it gets iteration from checkpoint_dict and starts incrementing from that number.
> 
> The coolest difference is that warm start can ignore embedding weights of pretrained model/checkpoint. So that anyone can warm start from [this tacotron 2 model](https://drive.google.com/file/d/1c5ZTuT7J08wLUoVZ2KkUs_VdZuJ86ZqA/view?usp=sharing) and get alignment image even after 500 iterations for any language. Thank you @rafaelvalle !

So, it mean I can get the English pretrained model of Tacotron2, and use warm start to train it for another language, even if I change alphabet, symbols, .... ?

**Comment by ksaidin** [comment]
you have to change alphabet, symbols and refer to basic cleaners from hparam settings. 
--warm_start parameter excludes embedding weights by default and if your dataset is OK, you can get more or less acceptable results in another language.

**Comment by rafaelvalle** [comment]
Closing due to inactivity.

**Comment by Ctibor67** [comment]
I still couldn't train my own set from scratch. So I tried to run 
_python train.py --output_directory=outdir --log_directory=logdir -c tacotron2_statedict.pt --warm_start_  
and train continues great. But now (30k iter) I need to edit the hparams (change learning rate). How do I continue training?
_python train.py --output_directory=outdir --log_directory=logdir -c checkpoint_30000.pt --warm_start_ 
or without _--warm_start_ ?

**Comment by EuphoriaCelestial** [comment]
> I still couldn't train my own set from scratch. So I tried to run
> _python train.py --output_directory=outdir --log_directory=logdir -c tacotron2_statedict.pt --warm_start_
> and train continues great. But now (30k iter) I need to edit the hparams (change learning rate). How do I continue training?
> _python train.py --output_directory=outdir --log_directory=logdir -c checkpoint_30000.pt --warm_start_
> or without _--warm_start_ ?

without `warm_start`
`python train.py --output_directory=outdir --log_directory=logdir -c checkpoint_30000.pt`
